torch/_inductor/scheduler.py

```diff
@@ -583,6 +582,12 @@ def can_inplace(self, read_dep: dependencies.MemoryDep):
 
         if len(self.read_writes.writes) == 1:
             write_dep = next(iter(self.read_writes.writes))
+            if read_dep.has_unbacked_symbols():
+                # can't inplace, as can't guarantee the exact size match
+                return False
+            if write_dep.has_unbacked_symbols():
+                # can't inplace, as can't guarantee the exact size match
+                return False
```

Comment:
We can probably make this a little better by doing an exact symbolic match, but this is fine for now.

Reply:
Actually, if we take symbolic product of the `read_dep` and `write_dep` shapes with unbacked SymInts in them, and those two symbolic products turn out to be equal, this means that the buffer sizes are provably the same and we can allow in-placing? I this is the case, I think, it's not too hard to do.

On the other hand, I have doubts about the current model for backed SymInts. We're relying on the size hints of the numels, but size hints being the same doesn't necessarily mean that the shapes are the same? I guess, matching symbolic numels would be more reliable?

Reply:
> Actually, if we take symbolic product of the read_dep and write_dep shapes with unbacked SymInts in them, and those two symbolic products turn out to be equal, this means that the buffer sizes are provably the same and we can allow in-placing? I this is the case, I think, it's not too hard to do.

Exactly.

> On the other hand, I have doubts about the current model for backed SymInts. We're relying on the size hints of the numels, but size hints being the same doesn't necessarily mean that the shapes are the same? I guess, matching symbolic numels would be more reliable?

Yeah, this seems a bit sus. Suppose that I have s0 and s1, and they happen to be the same this run. Why would reusing the buffer be OK if s0 and s1 later vary? Maybe we can construct a test case that forces a failure here...